### ISSUE TITLE
Corrupt callstack while inside ISPC functions

### ISSUE THREAD

**Issue by ecatchpole** [ispc/ispc#2345]
I'm doing some optimisations by replacing C++ with ISPC, and am finding it very difficult to use sampling profilers like Visual Studio's built-in CPU profiler, or third-party products such as Superluminal.

The timings surface in the wrong place in the call tree, and I believe it's because the callstack is corrupt while inside the ISPC function, resulting in samples being stored with a bad hierarchy behind them.

I'm using MSVC 16.11, ISPC 1.18.0.

You can see this when stepping through a simple contrived function such as:

```
// Bug.ispc
export uniform float BugFunction(uniform float x, uniform float y)
{
	uniform float xCos = cos(x);
	uniform float yCos = cos(y);
	return xCos + yCos;
}
```

```
// main.cpp
#include "bug.ispc.generated.h"

int main(int argc, char** argv)
{
	float result = ispc::BugFunction(1.0f, 2.0f);
	return (int)result;
}
```

In this gif, I am simply stepping through the program with F11. You can see that the callstack view on the right loses most entries once inside BugFunction.

![ISPCBacktraceBug](https://user-images.githubusercontent.com/96117112/179176565-45dd65f0-7c65-4bac-a86c-8feb35d90df2.gif)

I've not yet found any combination of ISPC parameters that fixes this. The gif above was produced using:
`"C:\Program Files\ISPC\ispc-v1.18.0-windows\bin\ispc.exe" %(FullPath) -o $(IntermediateOutputPath)%(Identity).obj -h $(ProjectDir)%(Identity).generated.h -g -O1 --target=avx2-i32x8 --no-omit-frame-pointer`

In the disassembly view, you can see a bit more of what's going on - the callstack corrupts on the very first instruction - `sub rsp 88h`, and is restored on the final `add rsp 88h` before the `ret`.

![image](https://user-images.githubusercontent.com/96117112/179178056-36757f09-25ce-4214-b433-a93e73f8749c.png)

A more elaborate contrived function to show up the problem in the VS profiler (as the previous example is too quick for a sampling profiler):

```
// Bug.ispc

export uniform float BugFunction(uniform float x, uniform float y)
{
	uniform float total = 0;
	for (int64 i = 0; i < 10000000; ++i)
		total += cos(x);
	return total;
}
```

```
// main.cpp
#include <cmath>
#include <ctime>
#include <cstdint>
#include "bug.ispc.generated.h"

// C++ version of ispc::BugFunction
float BugFunction(float x, float y)
{
	float total = 0;
	for (int64_t i = 0; i < 1000000; ++i)
		total += cos(x);
	return total;
}

float DoBugFunction()
{
	float total = (float)rand();
	for (int i = 0; i < 1000; ++i)
		total = cos(total);
	return ispc::BugFunction(1.0f, total); // Swap for just BugFunction(1.0f, total) to see the "working" profile
}

int main(int argc, char** argv)
{
	std::srand((unsigned int)std::time(nullptr));
	float result = 0;
	for (int i = 0; i < 1000; ++i)
		result += DoBugFunction();
	return (int)result;
}
```
Note that ispc::BugFunction shows up under a "[Unwalkable]" heading in the wrong place in the call tree
![image](https://user-images.githubusercontent.com/96117112/179180875-1bc65982-46dd-49e8-81a7-e1fbbb69483f.png)

This is how I would expect it to look (as if it were written in pure C++):
![image](https://user-images.githubusercontent.com/96117112/179180639-fea0736f-732f-4998-a8a5-9733b12f60e4.png)

**Comment by pbrubaker** [comment]
I think this could be related to #2322.

**Comment by pema99** [comment]
FWIW I've had better luck debugging and troubleshooting with SSE rather than AVX. Anecdotal, though.

**Comment by ecatchpole** [comment]
I've had a quick look - I get similar results with both SSE2 and AVX2. The size of the `sub rsp` operand changes, but the effect is the same.

I've also had a look at a similar function written in pure C++ - the `sub rsp` instruction is still there, but the call-stack is maintained when you step over it, leading me to think that this is purely an issue with symbols generation.

**Comment by nurmukhametov** [comment]
My conclusion that the corrupted callstack was caused by the lack of unwind info generated by `ISPC` on windows. The fix is here #2630 . It would be good if you can test it especially with profiling. I only tested that this fix is enough to see the sane callstack in the debugger.
